Synthesis.Bethesda/CLI/RunSynthesisPatcher.cs

```diff
@@ -24,14 +24,22 @@ public class RunSynthesisPatcher
         [Option('l', "LoadOrderFilePath", Required = false, HelpText = "Path to the load order file to use.")]
         public string LoadOrderFilePath { get; set; } = string.Empty;
 
+        [Option('s', "StatePath", Required = false, HelpText = "Path to the shared FormKey allocation state")]
```

Comment:
This actually brings up a larger discussion.

Synthesis is obviously mostly used for Mutagen patchers.  But it leaves open the potential for other programs to be used in its pipeline, if they are able to respond to this CLI command.  zEdit could be able to start up, see "run-patcher" was passed to it as args, and run its suite with the given parameters, and then synth could continue on with other patchers.

The introduction of this argument/concept, though, sort of breaks that compatibility.  All the sudden, we're passing zedit a path to a formkey persistence file that it has no clue how to parse.  Everything else up until this point has been concepts everyone agrees on -> Bethesda formats.  The persistence object is 100% a Mutagen-only concept/format. 

Yeah, zedit/mator could learn to parse our persistence systems, but it's definitely not a clean generic API anymore -> it requires coordination with Mutagen defined formats.  Before it was an easy as of "hey, just run your stuff, and put your results here, please".  Now it's a "and also parse this custom persistence object with this specific format we made up".

Not sure how to reconcile that, just yet.